controller/users.go

```diff
@@ -157,6 +193,137 @@ func (c *UsersController) getKeycloakProfileInformation(ctx context.Context, tok
 	return response, err
 }
 
+func (c *UsersController) updateUserInDB(id *uuid.UUID, ctx *app.UpdateUserAsServiceAccountUsersContext) error {
+
+	// We'll refactor the old Users update API to be redirected to Auth service in the near future.
+	// Hence, not spending time in refactoring that to consume this function.
+
+	returnResponse := application.Transactional(c.db, func(appl application.Application) error {
+		identity, err := appl.Identities().Load(ctx, *id)
+		if err != nil || identity == nil {
+			log.Error(ctx, map[string]interface{}{
+				"identity_id": id,
+			}, "auth token contains id %s of unknown Identity", *id)
+			jerrors, _ := jsonapi.ErrorToJSONAPIErrors(ctx, goa.ErrUnauthorized(fmt.Sprintf("Auth token contains id %s of unknown Identity\n", *id)))
+			return ctx.Unauthorized(jerrors)
+		}
+
+		var user *account.User
+		if identity.UserID.Valid {
+			user, err = appl.Users().Load(ctx.Context, identity.UserID.UUID)
+			if err != nil {
+				return jsonapi.JSONErrorResponse(ctx, errs.Wrap(err, fmt.Sprintf("Can't load user with id %s", identity.UserID.UUID)))
+			}
+		}
+
+		updatedEmail := ctx.Payload.Data.Attributes.Email
+		if updatedEmail != nil && *updatedEmail != user.Email {
+			isValid := isEmailValid(*updatedEmail)
+			if !isValid {
+				jerrors, _ := jsonapi.ErrorToJSONAPIErrors(ctx, goa.ErrInvalidRequest(fmt.Sprintf("invalid value assigned to email for identity with id %s and user with id %s", identity.ID, identity.UserID.UUID)))
+				return ctx.BadRequest(jerrors)
+			}
+			isUnique, err := isEmailUnique(appl, *updatedEmail, *user)
+			if err != nil {
+				return jsonapi.JSONErrorResponse(ctx, errs.Wrap(err, fmt.Sprintf("error updating identitity with id %s and user with id %s", identity.ID, identity.UserID.UUID)))
+			}
+			if !isUnique {
+				jerrors, _ := jsonapi.ErrorToJSONAPIErrors(ctx, goa.ErrInvalidRequest(fmt.Sprintf("email address: %s is already in use", *updatedEmail)))
+				return ctx.Conflict(jerrors)
+			}
+			user.Email = *updatedEmail
+		}
+
+		updatedUserName := ctx.Payload.Data.Attributes.Username
+		if updatedUserName != nil && *updatedUserName != identity.Username {
+			isValid := isUsernameValid(*updatedUserName)
+			if !isValid {
+				jerrors, _ := jsonapi.ErrorToJSONAPIErrors(ctx, goa.ErrInvalidRequest(fmt.Sprintf("invalid value assigned to username for identity with id %s and user with id %s", identity.ID, identity.UserID.UUID)))
+				return ctx.BadRequest(jerrors)
+			}
+			if identity.RegistrationCompleted {
```

Comment:
We don't care if registration complete or not here, do we? It's up to Auth to decide when it needs to update the username.

Reply:
Correct.. this endpoint should not have business logic.. should be plain persistence.

Reply:
removed a ton of business logic https://github.com/fabric8-services/fabric8-wit/pull/1637/commits/fccb1b01c0b946bfeb2d8be8e4e948f06f7ac45f#diff-da062b312519ee2a61dd669f1ec964d4